addons/website_slides/models/slide_channel.py

```diff
@@ -454,6 +458,36 @@ def _get_default_enroll_msg(self):
     def _compute_enroll(self):
         self.filtered(lambda channel: channel.visibility == 'members').enroll = 'invite'
 
+    @api.depends('visibility', 'is_member')
+    @api.depends_context('uid')
+    def _compute_is_visible(self):
+        for channel in self:
+            channel.is_visible = (
+                channel.visibility == 'public'
+                or channel.is_member
+                or (not self.env.user._is_public() and channel.visibility == 'connected')
+            )
+
+    @api.model
+    def _search_is_visible(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise NotImplementedError('Operation not supported')
+        check_is_visible = value if operator == '=' else not value
+        if check_is_visible:
+            domain = ['|', ('is_member', '=', True)] + (
+                [('visibility', '=', 'public')]
+                if self.env.user._is_public()
+                else [('visibility', 'in', ['public', 'connected'])]
+            )
+        else:
+            domain = ['|', ('is_member', '=', False)] + (
```

Comment:
shouldn't be "and" instead of "or" ?
(some test are missing if it wasn't catched)